src/inverse-kinematics/include/iDynTree/InverseKinematics.h

```diff
@@ -356,12 +356,35 @@ class iDynTree::InverseKinematics
                                     const iDynTree::Transform& constraintValue);
 
     /*!
-     * Add a constant inequality constraint on the projection of the center of mass,
-     * assuming one support links.
+     * Activate a given constraint previously added with an addFrame**Constraint method.
+     *
+     * \note In this version of iDynTree, it is not possible to change the nature of the constraint
+     *       (Full, Position or Rotation) when activating it again.
      *
-     * This method assume that the position of two links is constrained by a FrameConstraint,
-     * and adds a inequality constraint to ensure that the center of mass projection lies on the
-     * convex hull of the contact polygon.
+     * @note This method returns true even if the frame constraint was already activate, it only
+     *       returns false if the constraint was never added.
+     * @warning This method is not meant to be called at each IK loop, and it can increase the computational
+     *          time of the next call to solve.
+     *
+     * @param frameName       the name of the frame on which to attach the constraint
+     * @param newConstraintValue the pose of the constrained frame (r) in the world frame (w), i.e. ʷHᵣ .
+     * @return true if successful, false otherwise.
+     */
+    bool activateFrameConstraint(const std::string& frameName,
```

Comment:
I don't remember if the constraint class is internal or publicly available.
In the latter, what about the following method
```
Constraint& constraintForFrameName(const std::string& frameName);
```
and then we can call on it the set active, etc?

Reply:
It is a bit more complicated, because when changing the state of a constraint we need to invalidate the cache of  the allocated buffers, see https://github.com/robotology/idyntree/pull/389/files#diff-681a27f55b063d58c7314c217b6e21a9R230 . When we will have a proper constraint class this will definitely be a feature to implement.